crates/pallet-domains/src/lib.rs

```diff
@@ -2370,13 +2403,44 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
+    fn check_execution_receipt_version(
```

Comment:
Please add comments to new methods that are missing them. If a method has been substantially modified, it would also be helpful to add a short comment.